Makes the plugin package discoverable

Voldivh opened the pull request.

Voldivh committed:
New package (not discoverable yet)

Signed-off-by: Voldivh <eloyabmfcv@gmail.com>

Voldivh committed:
Removes files from folder

Signed-off-by: Voldivh <eloyabmfcv@gmail.com>

Voldivh committed:
Makes the plugin discoverable

Signed-off-by: Voldivh <eloyabmfcv@gmail.com>

Voldivh committed:
Merge remote-tracking branch 'origin/main' into voldivh/package_creation

Signed-off-by: Voldivh <eloyabmfcv@gmail.com>

francocipollone reviewed (commented):
Before diving into review
1. Did you find which was the cause of the plugin not being discoverable?
2. Note that there are some changes that I think aren't related to the PR's purpose(at least pr's title.)
   - Linters being added
   - resources being added.

Voldivh commented:
> 1. Did you find which was the cause of the plugin not being discoverable?

Yes, it appers there were 2 simple things that caused troubles:
1. On the package.xml we defined the licence by doing `<license file='LICENSE'> BSD 3 Clause </license>` and when I was testing the package it gave me problem the `file='LICENSE'` so I removed it.
2. I noticed that on the setup.cfg the `[develop]` and `[install]` were using snake_case ('_') instead of hyphen ('-') on the naming of the variables (I'm not sure this actually makes a difference but I left it with hyphen since that is how it is created when doing `ros2 pkg create --build-type ament_python <package_name>`).

> 2. Note that there are some changes that I think aren't related to the PR's purpose(at least pr's title.)
>    
>    * Linters being added
>    * resources being added.
You are right, I forgot to delete a couple of things. For the resource I changed the name of the .ui file to keep consistency with other rqt plugins.

Voldivh committed:
Deletes unnecesary files for PR

Signed-off-by: Voldivh <eloyabmfcv@gmail.com>

francocipollone commented on package.xml:
```diff
@@ -1,12 +1,13 @@
 
-  <maintainer email="eloy.briceno@ekumenlabs.com">Eloy Briceno</maintainer>
-
-  <license file="LICENSE">BSD Clause 3</license>
```
Revert this change, It shouldn't be related.

PS: I tested locally just in case and it doesn't seem to affect.

Voldivh replied:
Done.

francocipollone commented on resource/gauges_2.ui:
```diff
@@ -29,4 +29,4 @@
  <resources/>
  <connections/>
-</ui>
+</ui>
```
Small unintended change

Voldivh replied:
So there is no change to be done from what I understand.

francocipollone replied:
Check the tab where you review the pr
![image](https://user-images.githubusercontent.com/53065142/192590696-d375d52c-7f13-4923-8c00-35ae5c16ea64.png)

Voldivh replied:
Oh I see.

francocipollone commented on setup.py:
```diff
@@ -11,16 +11,19 @@
         'rqt_gauges_2 is a Python GUI plugin providing a visualization tool for several sensors.'
     ),
+    license='BSD Clause 3',
+    tests_require=['pytest'],
```
Don't belongs to this PR.

Voldivh replied:
Done.

francocipollone commented on setup.py:
```diff
@@ -11,16 +11,19 @@
         ('share/' + package_name, ['plugin.xml']),
     ],
     install_requires=['setuptools'],
+    zip_safe=True,
```
What is this for?

Voldivh replied:
I'm not sure, it came as default with the command `ros2 pkg create --build-type ament_python <package_name>` so i left it.

francocipollone replied:
https://setuptools.pypa.io/en/latest/deprecated/zip_safe.html
Just delete it

Voldivh replied:
Makes sense. Done.

francocipollone reviewed (changes requested):
LGTM, just some comments

Voldivh committed:
Deletes unnecesary flags and adds the licence file to the package.xml

Signed-off-by: Voldivh <eloyabmfcv@gmail.com>

francocipollone reviewed (approved):
LGTM
